ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ServerVerticle.java

```diff
@@ -219,23 +219,31 @@ private void setupAuthHandlers(final Router router) {
     }
   }
 
-  // If we have a Jaas handler configured and we have Basic credentials then we should auth
-  // with that
   private static void wrappedAuthHandler(final RoutingContext routingContext,
-      final Optional<Handler<RoutingContext>> jaasAuthHandler,
+      final Optional<AuthHandler> jaasAuthHandler,
       final Optional<Handler<RoutingContext>> pluginHandler) {
     if (jaasAuthHandler.isPresent()) {
+      // If we have a Jaas handler configured and we have Basic credentials then we should auth
+      // with that
       final String authHeader = routingContext.request().getHeader("Authorization");
-      if (authHeader == null || authHeader.toLowerCase().startsWith("basic ")) {
+      if (authHeader != null && authHeader.toLowerCase().startsWith("basic ")) {
         jaasAuthHandler.get().handle(routingContext);
         return;
       }
     }
-    pluginHandler
-        .ifPresent(routingContextHandler -> routingContextHandler.handle(routingContext));
+    // Fall through to authing with any authentication plugin
+    if (pluginHandler.isPresent()) {
+      pluginHandler.get().handle(routingContext);
```

Comment:
To confirm my understanding - this is what would handle authenticating/authorizing alternative credentials? (e.g. a bearer token like ccloud supports)

Reply:
Yes